2_process/src/bind_data.R

```diff
@@ -0,0 +1,17 @@
+bind_data <- function(eval_1_data, eval_2_data, eval_3_data, eval_4_data, search_string, file_outpath) {
+  
+  # Function to bind downloaded output files together
+  
+    # assume common out directory among files
+    out_dir <- sub("(/.*?/).*", "\\1", eval_1_data)
+  
+    # read in all files in out directory matching the search string and combine
+    filenames <- list.files(out_dir, pattern = paste0(search_string), full.names = TRUE)
```

Comment:
I think this can be simplified because the values of the incoming `site_1_data`, `site_2_data`, etc are already the full filenames. You can also allow this to scale in case we add or subtract sites by using `...` as an argument. For example,

```r
bind_data <- function(file_outpath, ...) {
    all_files <- list(...)
    combined_df <- lapply(all_files, readr::read_csv) %>% bind_rows()
    readr::write_csv(combined_df, file_outpath)
    return(file_outpath)
}
```

Then, you could call it in your pipeline with 

```r
tar_target(
    compiled_nwis_data,
    bind_data(
        "2_process/out/all_nwisdata.csv", # file outpath
        site_1_data,
        site_2_data,
        site_3_data,
        site_4_data
    ),
    format = "file"
)
```

Reply:
This is a great trick! I didn't realize this could be done.